src/App/Game/GameModePicker.tsx

```diff
@@ -144,6 +152,33 @@ export class GameModePicker extends React.Component<IGameModePickerProps, IGameM
             />
           </div>
         );
+      } else if (extraInfo.type === 'dropdown') {
+        const dropdown = (extraInfo as IGameModeExtraInfoDropdown);
+        const list: JSX.Element[] = dropdown.options.map((option, idx) => {
+          idx++;
+          return (
+            <MenuItem
+              onClick={this._handleExtraInfoChange(info.mode, idx)}
+              key={option}
+              value={option}
+              selected={this._getExtraInfoValue(info) === idx}
+            >
+              {option}
+            </MenuItem>);
+        });
+        return (
+          <div style={{ marginTop: '16px' }}>
+            <MenuList
+              style={{
+                paddingRight: '20%',
+                display: 'flex',
+                listStyle: 'none',
+              }}
+            >
+              {list}
+            </MenuList>
+          </div>
```

Comment:
this function is too big, can you break it down into smaller ones ? ie getDropDown and getSlider ?